packages/react/src/components/Popover/index.tsx

```diff
@@ -0,0 +1,287 @@
+import React, { useState, useEffect, ReactNode, forwardRef, Ref } from 'react';
+import { createPortal } from 'react-dom';
+import { useId } from 'react-id-generator';
+import { Placement } from '@popperjs/core';
+import { usePopper } from 'react-popper';
+import { isBrowser } from '../../utils/is-browser';
+import { Cauldron } from '../../types';
+
+import classnames from 'classnames';
+import ClickOutsideListener from '../ClickOutsideListener';
+import Button from '../Button';
+import FocusTrap from 'focus-trap-react';
+import focusableSelector from '../../utils/focusable-selector';
+import AriaIsolate from '../../utils/aria-isolate';
+import useSharedRef from '../../utils/useSharedRef';
+
+export type PopoverVariant = 'alert' | 'custom';
+
+type BaseProps = React.HTMLAttributes<HTMLDivElement> & {
+  target: React.RefObject<HTMLElement> | HTMLElement;
+  variant?: PopoverVariant;
+  show: boolean;
+  onClose: () => void;
+  placement?: Placement;
+  portal?: React.RefObject<HTMLElement> | HTMLElement;
+};
+
+type CustomProps = BaseProps & {
+  variant: 'custom';
+  applyButtonText?: string;
+  onApply?: () => void;
+  closeButtonText?: string;
+  infoText?: string;
+  chidlren: ReactNode;
+} & Cauldron.LabelProps;
+
+type AlertProps = BaseProps & {
+  variant: 'alert';
```

Comment:
I'm thinking we should call this something else, since "alert" has a different connotation with regards to accessibility. I wonder if something like `prompt` might be better? What do you think?